src/Symfony/Bundle/FrameworkBundle/EventListener/SuggestMissingPackageSubscriber.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\EventListener;
+
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleErrorEvent;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Suggests a package, that should be installed (via composer),
+ * if the package is missing, and the input command namespace can be mapped to a Symfony bundle.
+ *
+ * @author Przemysław Bogusz <przemyslaw.bogusz@tubotax.pl>
+ *
+ * @internal
+ */
+final class SuggestMissingPackageSubscriber implements EventSubscriberInterface
+{
+    private const PACKAGES = [
+        'doctrine' => [
+            'fixtures' => ['DoctrineFixturesBundle', 'doctrine/doctrine-fixtures-bundle --dev'],
+            'mongodb' => ['DoctrineMongoDBBundle', 'doctrine/mongodb-odm-bundle'],
+            '_default' => ['Doctrine ORM', 'symfony/orm-pack'],
+        ],
+        'generate' => [
+            '_default' => ['SensioGeneratorBundle', 'sensio/generator-bundle'],
+        ],
+        'make' => [
+            '_default' => ['MakerBundle', 'symfony/maker-bundle --dev'],
+        ],
+        'server' => [
+            'dump' => ['VarDumper Component', 'symfony/var-dumper --dev'],
+            '_default' => ['WebServerBundle', 'symfony/web-server-bundle --dev'],
+        ],
+    ];
+
+    public function onConsoleError(ConsoleErrorEvent $event): void
+    {
+        if (!$event->getError() instanceof CommandNotFoundException || empty($input = $event->getInput()->getFirstArgument())) {
+            return;
+        }
+
+        $input = explode(':', $input);
+        $namespace = $input[0];
+        $command = $input[1] ?? '';
+
+        if (!isset(self::PACKAGES[$namespace])) {
+            return;
+        }
+
+        if (isset(self::PACKAGES[$namespace][$command]) && '' !== $command) {
+            $suggestion = self::PACKAGES[$namespace][$command];
+            $exact = true;
+        } else {
+            $suggestion = self::PACKAGES[$namespace]['_default'];
+            $exact = false;
+        }
+
+        $error = $event->getError();
+
+        if (!empty($error->getAlternatives()) && !$exact) {
```

Comment:
Same here, you can just remove `empty`

Reply:
This one is for an array. I can use `count`, if it's preferable. If I don't check that the array has any elements, the method will not return a proper message.

For example, on running `php bin/console doctrine` it will just state `Command "doctrine" is not defined.` instead of `Command "doctrine" is not defined. You may be looking for a command provided by "Doctrine ORM" which is currently not installed. Try running "composer require symfony/orm-pack". `

Reply:
` if ($error->getAlternatives() && !$exact) {` would just work, no need for anything around.

Reply:
Done